### PR TITLE
PCHR-2527: Make calendar show requests touching different months

### PR DESCRIPTION
**Pull request by igorpavlov**
## Overview

Currently the Leave Calendar does not reflect leave requests that have start and end dates in different months.

## Technical Details

Previously, only requests that had the both start and end dates from the specified month are shown.

```js
from_date in January AND to_date in January
```

Now, all requests that have at least one day from the specified months are shown.

```js
from_date in January OR to_date in January
```

## Comments

1. The additional check is needed since in this PR the Calendar Month component now gets requests that might touch days from other months. These days should not be considered since they are not going to be rendered.

```js
while (pointerDate.isSameOrBefore(toDate)) {
  // Ensure that pointerDate is in same month/year that component represents
  if (pointerDate.month() === vm.month.index && pointerDate.year() === vm.month.year) {
    days.push(_.find(vm.month.days, function (day) {
      return day.date === pointerDate.format('YYYY-MM-DD');
    }));
  }

  pointerDate.add(1, 'day');
}
```

2. Please see this PR https://github.com/civicrm/civicrm-core/pull/9844 for the reference to the OR operator support in CiviCRM.

⚠️ This PR can only be merged after https://github.com/civicrm/civihr/pull/2085 is merged and synced.

---

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js [hunk 1]
```diff
@@ -315,9 +314,13 @@ define([
      * @return {Promise}
      */
     function loadMonthLeaveRequests () {
+      var range = { from: vm.month.days[0].date,
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: The indentation looks weird, please make it,

```
var range = { 
  from: vm.month.days[0].date,
  to: vm.month.days[vm.month.days.length - 1].date 
};
```

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js [hunk 2]
```diff
@@ -315,9 +314,13 @@ define([
...
+        to: vm.month.days[vm.month.days.length - 1].date };
+
       return LeaveRequest.all({
-        from_date: { from: vm.month.days[0].date },
-        to_date: { to: vm.month.days[vm.month.days.length - 1].date },
+        from_date: range,
+        to_date: range,
+        options: { or: [['from_date', 'to_date']] },
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: Here we have double square brackets, please confirm if that is correct?

Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: Please leave a comment in the code, it might get confusing in future.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-calendar-month.component.spec.js [hunk 3]
```diff
@@ -149,13 +149,14 @@
               }));
             });
 
-            it("uses the selected months' first and last day as date delimiters", function () {
+            it('loads all requests touching the specified month', function () {
               var month = controller.month;
+              var range = { from: month.days[0].date,
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-calendar-month.component.spec.js]: The indentation looks weird, please fix it as suggested above.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js [hunk 4]
```diff
@@ -175,9 +173,7 @@ define([
         });
       });
 
-      deferred.resolve();
-
-      return deferred.promise;
+      $q.resolve();
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: Should not it be `return $q.resolve()`?

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js [hunk 5]
```diff
@@ -248,9 +244,12 @@ define([
       var toDate = moment(leaveRequest.to_date);
 
       while (pointerDate.isSameOrBefore(toDate)) {
-        days.push(_.find(vm.month.days, function (day) {
-          return day.date === pointerDate.format('YYYY-MM-DD');
-        }));
+        // Ensure that pointerDate is in same month/year that component represents
+        if (pointerDate.month() === vm.month.index && pointerDate.year() === vm.month.year) {
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: Why this change is necessary? Please document it.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js (hunk 4 above)
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: Pardon.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js (hunk 5 above)
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: @deb1990 do you suggest to extend the comment or document it in the PR description?

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js (hunk 1 above)
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: Done.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js (hunk 2 above)
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: It is correct. cc @AkA84

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js (hunk 5 above)
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js]: The comment in the code makes sense, but I am not sure why it was necessary in the context of the ticket, so if you document it in the PR, it will be helpful.
